### DISCUSSION TITLE
Cannot change legend font size for rdMolDraw2D

### DISCUSSION THREAD

**Discussion by and-tos** [rdkit/rdkit#5529] [Q&A] [answered]
In the blogpost example described [here](https://greglandrum.github.io/rdkit-blog/tutorial/prototypes/drawing/2021/08/07/rgd-and-highlighting.html) I cannot adjust the legend fontsize.
I tried e.g. inserting `dos.legendFontSize = 300` but it has no effect on the plot, whereas e.g. `dos.minFontSize = 300` will result in a changed fontsize for the atom symbols. Any idea what might be going wrong?

```
def highlight_rgroups(mol,row,core,width=350,height=200,
                      fillRings=True,legend="",
                      sourceIdxProperty="SourceAtomIdx",
                      lbls=('R1','R2','R3','R4')):
   ...
  d2d = rdMolDraw2D.MolDraw2DCairo(width,height)
  
  dos = d2d.drawOptions()
  dos.useBWAtomPalette()
  # try changing legened font size
  d2d.SetFontSize(300)
  dos.legendFontSize = 300
   ...
```

rdkit.__version__
'2022.03.5'

**Comment by LukaThorn** [comment]
The only solution I have found for adjusting the legend font size is to change the `legendFraction`. This works for my setup below, but I did not test it with yours. I also tried some slightly different syntax for the legend _font size_, but still had no luck there.

```
from rdkit import Chem
from rdkit.Chem import Draw
from rdkit.Chem.Draw import rdMolDraw2D
import cairosvg


def molecule_to_pdf(mol, file_name, width=300, height=100):
    """Save substance structure as PDF"""

    # Define full path name
    full_path = f"./figs/2Dstruct/2D_{file_name}.pdf"

    # Pull molecule label data
    id = mol.GetProp("DTXSID")
    prefname = mol.GetProp("PREFERRED_NAME")
    legend = f"{id}\n{prefname}"

    # Render high resolution molecule
    drawer = rdMolDraw2D.MolDraw2DSVG(width, height)
    # drawer.drawOptions().fixedFontSize = 16  # atom font size alternative
    drawer.drawOptions().minFontSize = 12  # atom font size alternative
    drawer.SetLineWidth(1)  # bond line width

    # Attempts at legend customization
    drawer.drawOptions().legendFraction = 0.3  ##### WORKING WORK-AROUND #####
    # drawer.drawOptions().legendFontSize = 14  ##### still doesn't work #####
    drawer.DrawMolecule(mol, legend=legend)

    drawer.FinishDrawing()

    # Export to pdf
    cairosvg.svg2pdf(bytestring=drawer.GetDrawingText().encode(), write_to=full_path)
```

**Comment by DavidACosgrove** [comment, marked as answer]
Whatever size you set the legend font to, it will always be reduced so that
the whole legend fits in the image. By default, the bottom 10th of the
image is allocated to the legend if there is one. As you have found,
increasing legendFraction gives more space over to the legend and this
allows a larger font if width is not the limiting factor. The legend can be
multiple lines, so if you add newlines to the string at appropriate places
you can make it taller and shorter which might help.

Dave


On Thu, 15 Sep 2022 at 01:57, Luka Thornton ***@***.***>
wrote:

> The only solution I have found for adjusting the legend font size is to
> change the legendFraction. This works for my setup below, but I did not
> test it with yours. I also tried some slightly different syntax for the
> legend *font size*, but still had no luck there.
>
> from rdkit import Chem
> from rdkit.Chem import Draw
> from rdkit.Chem.Draw import rdMolDraw2D
> import cairosvg
>
>
> def molecule_to_pdf(mol, file_name, width=300, height=100):
>     """Save substance structure as PDF"""
>
>     # Define full path name
>     full_path = f"./figs/2Dstruct/2D_{file_name}.pdf"
>
>     # Pull molecule label data
>     id = mol.GetProp("DTXSID")
>     prefname = mol.GetProp("PREFERRED_NAME")
>     legend = f"{id}\n{prefname}"
>
>     # Render high resolution molecule
>     drawer = rdMolDraw2D.MolDraw2DSVG(width, height)
>     # drawer.drawOptions().fixedFontSize = 16  # atom font size alternative
>     drawer.drawOptions().minFontSize = 12  # atom font size alternative
>     drawer.SetLineWidth(1)  # bond line width
>
>     # Attempts at legend customization
>     drawer.drawOptions().legendFraction = 0.3  ##### WORKING WORK-AROUND #####
>     # drawer.drawOptions().legendFontSize = 14  ##### still doesn't work #####
>     drawer.DrawMolecule(mol, legend=legend)
>
>     drawer.FinishDrawing()
>
>     # Export to pdf
>     cairosvg.svg2pdf(bytestring=drawer.GetDrawingText().encode(), write_to=full_path)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/5529#discussioncomment-3649073>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACGF2FUDVL2AAFHXGD3KHODV6JQ7FANCNFSM6AAAAAAQBZEZAI>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
David Cosgrove
Freelance computational chemistry and chemoinformatics developer
http://cozchemix.co.uk
